workflow/controller/dag.go

```diff
@@ -524,6 +524,14 @@ func (woc *wfOperationCtx) executeDAGTask(ctx context.Context, dagCtx *dagContex
 				return
 			}
 		}
+		if node.Completed() {
+			// if the node type is retry, and its last children is completed, it will be completed after woc.executeTemplate;
+			hasOnExitNode, onExitNode, err := woc.runOnExitNode(ctx, task.GetExitHook(woc.execWf.Spec.Arguments), node, dagCtx.boundaryID, dagCtx.tmplCtx, "tasks."+taskName)
+			if hasOnExitNode && (onExitNode == nil || !onExitNode.Fulfilled() || err != nil) {
+				// The onExit node is either not complete or has errored out, return.
+				return
+			}
+		}
 	}
```

Comment:
sorry for nitpick, but I think on line 528 maybe the comment should say "its last _child_ is completed"?

Reply:
done